adobe-sign-services.module/adobe-sign-services.core/src/main/java/com/aftia/adobe/adobesign/core/services/agreement/Agreement.java

```diff
@@ -26,4 +25,23 @@ public static String generateAgreement(AdobeSignApiClient apiClient, AdobeSignAu
 
     }
 
+    public static String generateAgreementView(AdobeSignApiClient apiClient, AdobeSignAuthentication authentication,
+                                               String agreementId, AgreementViewInfo agreementViewInfo, String xApiUser,
+                                               String xOnBehalfOfUser)
+            throws AdobeSignServiceException {
+        try {
+            AgreementsApi agreementsApi = new AgreementsApi(apiClient.buildClient());
+            AgreementViews agreementViews = agreementsApi.createAgreementView(
+                    String.format("Bearer %s", authentication.authenticate()),
+                    agreementId,
+                    agreementViewInfo,
+                    xApiUser,
+                    xOnBehalfOfUser);
+            return agreementViews.getAgreementViewList().get(0).getUrl();
```

Comment:
If there is a typo in the agreementId this will generate a NPE maybe check the list is not null and log a ERROR or WARN to help the user debug